src/terminal_pane/terminal_pane.rs

```diff
@@ -164,6 +164,48 @@ impl TerminalPane {
         self.reflow_lines();
         self.mark_for_rerender();
     }
+    pub fn is_directly_right_of(&self, other: &TerminalPane) -> bool {
+        self.get_x() == other.get_x() + other.get_columns() + 1
+    }
+    pub fn is_directly_left_of(&self, other: &TerminalPane) -> bool {
+        self.get_x() + self.get_columns() + 1 == other.get_x()
+    }
+    pub fn is_directly_below(&self, other: &TerminalPane) -> bool {
+        self.get_y() == other.get_y() + other.get_rows() + 1
+    }
+    pub fn is_directly_above(&self, other: &TerminalPane) -> bool {
+        self.get_y() + self.get_rows() + 1 == other.get_y()
+    }
+    pub fn horizontally_overlaps_with(&self, other: &TerminalPane) -> bool {
+        (self.get_y() >= other.get_y() && self.get_y() <= (other.get_y() + other.get_rows()))
+            || ((self.get_y() + self.get_rows()) <= (other.get_y() + other.get_rows())
+                && (self.get_y() + self.get_rows()) > other.get_y())
+            || (self.get_y() <= other.get_y()
+                && (self.get_y() + self.get_rows() >= (other.get_y() + other.get_rows())))
+            || (other.get_y() <= self.get_y()
+                && (other.get_y() + other.get_rows() >= (self.get_y() + self.get_rows())))
+    }
+    pub fn get_horizontal_overlap_with(&self, other: &TerminalPane) -> usize {
+        std::cmp::min(
+            self.get_y() + self.get_rows(),
+            other.get_y() + other.get_rows(),
+        ) - std::cmp::max(self.get_y(), other.get_y())
+    }
+    pub fn vertically_overlaps_with(&self, other: &TerminalPane) -> bool {
+        (self.get_x() >= other.get_x() && self.get_x() <= (other.get_x() + other.get_columns()))
+            || ((self.get_x() + self.get_columns()) <= (other.get_x() + other.get_columns())
+                && (self.get_x() + self.get_columns()) > other.get_x())
+            || (self.get_x() <= other.get_x()
+                && (self.get_x() + self.get_columns() >= (other.get_x() + other.get_columns())))
+            || (other.get_x() <= self.get_x()
+                && (other.get_x() + other.get_columns() >= (self.get_x() + self.get_columns())))
+    }
+    pub fn get_vertical_overlap_with(&self, other: &TerminalPane) -> usize {
+        std::cmp::min(
+            self.get_x() + self.get_columns(),
+            other.get_x() + other.get_columns(),
+        ) - std::cmp::max(self.get_x(), other.get_x())
+    }
```

Comment:
Do you think we can implement these on the `Rect` trait instead of on terminal_pane? That way when we add other pane types (eg. plugins) they will get this for free.

Reply:
Yup, don't see why it can't be moved, will do